addons/sale_project_account/models/account_move.py

```diff
@@ -0,0 +1,13 @@
+# -*- coding: utf-8 -*-
+
+from odoo import models
+
+
+class AccountMoveLine(models.Model):
+    _inherit = 'account.move.line'
+
+    def _compute_analytic_account_id(self):
+        # when a project creates an aml, it adds an analytic account to it. the following filter is to save this
+        # analytic account from being overridden by analytic default rules and lack thereof
+        project_amls = self.filtered(lambda aml: aml.analytic_account_id and any(aml.sale_line_ids.project_id))
```

Comment:
Hello @momegahed 🙂 
I am wondering whether keeping the AA when changing the product_id makes really sense.
Shouldn't we rather remove the records where
```python
aml.product_id != aml._origin.product_id
```
 ?

Reply:
After having thought of it, I think that if we arrive that far in the flow, the invoice should keep the project AA.
You can ignore my previous comment.
Thanks for the fix 🙂